ethcore/sync/src/light_sync/mod.rs

```diff
@@ -213,6 +214,43 @@ enum SyncState {
 	Rounds(SyncRound),
 }
 
+/// A wrapper around the SyncState that makes sure to
+/// update the giving reference to `is_idle`
+struct SyncStateWrapper {
+	state: SyncState,
+}
+
+impl SyncStateWrapper {
+	/// Create a new wrapper for SyncState::Idle
+	pub fn idle() -> Self {
+		SyncStateWrapper {
+			state: SyncState::Idle,
+		}
+	}
+
+	/// Set the new state's value, making sure `is_idle` gets updated
+	pub fn set(&mut self, state: SyncState, is_idle_handle: &mut bool) {
+		*is_idle_handle = match state {
+			SyncState::Idle => true,
+			_ => false,
+		};
+		self.state = state;
+	}
+
+	/// Returns the internal state's value
+	pub fn get(self) -> SyncState {
```

Comment:
`get` usually takes a reference, for [Cell](https://doc.rust-lang.org/std/cell/struct.Cell.html#method.into_inner) this is called `into_inner`
```suggestion
	pub fn into_inner(self) -> SyncState {
```